tsp/src/store.rs

```diff
@@ -732,9 +745,15 @@ impl Store {
         sender: &str,
         receiver: &str,
     ) -> Result<(Url, Vec<u8>), Error> {
-        self.set_relation_status_for_vid(receiver, RelationshipStatus::Unrelated)?;
-
-        let thread_id = Default::default(); // FNORD
+        let old_relationship =
+            self.replace_relation_status_for_vid(receiver, RelationshipStatus::Unrelated)?;
+
+        let thread_id = match old_relationship {
+            RelationshipStatus::_Controlled => todo!(),
+            RelationshipStatus::Bidirectional { thread_id, .. } => thread_id,
+            RelationshipStatus::Unidirectional { thread_id } => thread_id,
+            RelationshipStatus::Unrelated => todo!(),
+        };
```

Comment:
so, what should the other branches do here? also are the bidirectional `outstanding_nested_thread_ids` relevant here?

Reply:
`Unrelated` and `_Controlled` -> there is no relationship to cancel, so return an Error like `Error::Relationship`. 

The `outstanding_nested_thread_ids` are unrelated here. Although this does raise the question of what it means if someone tries to request a new nested relationship, then cancels the parent relationship.